Sources/Network/RandomUrlLoadBalancer.swift

```diff
@@ -0,0 +1,29 @@
+//
+//  Copyright (c) 2020-2021 MobileCoin. All rights reserved.
+//
+
+//protocol UrlLoadBalancer {
+//    associatedtype AnyScheme: Scheme
+//    func getNextUrl() -> MobileCoinUrl<AnyScheme>
+//}
+
+
+class RandomUrlLoadBalancer<Url: MobileCoinUrlProtocol> {
+    var rng = SystemRandomNumberGenerator()
+
+    let urlsTyped: [Url]
+    private(set) var currentUrl: Url?
+    
+    init(urls: [Url]) throws {
+        guard !urls.isEmpty else {
+            throw InvalidInputError("urls array must not be empty")
+        }
+        self.urlsTyped = urls
+        _ = nextUrl()
+    }
+    
+    func nextUrl() -> Url? {
```

Comment:
add discardable result to nextUrl() , will let you call and ignore return value so line 22 would be "nextUrl()"

Reply:
Ah, nice... thanks!

Reply:
And I see what you mean about keeping the protocol as-is...  good point to minimize the impact.  Will ping you on Slack to discuss the details.